### PR TITLE
implemented liquibase, added delete, update, search methods

### PR DESCRIPTION
**Pull request by DmytroV95**

### CODE REVIEW

**Review by IvanFadieiev — APPROVED** [review]
Looks good to me! GJ!

File: src/main/java/com/varukha/onlinebookstore/repository/book/BookSpecificationBuilder.java [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+@Component
+public class BookSpecificationBuilder implements SpecificationBuilder<Book> {
+    public static final String TITLE_KEY = "title";
+    public static final String AUTHOR_KEY = "author";
+    private final SpecificationProviderManager<Book> bookSpecificationProviderManager;
+
+    @Override
+    public Specification<Book> build(BookSearchParametersDto searchParametersDto) {
+        Specification<Book> spec = Specification.where(null);
+        if (searchParametersDto.titles() != null
```
Comment (vovasalyha) [inline src/main/java/com/varukha/onlinebookstore/repository/book/BookSpecificationBuilder.java:21]: you are duplicating the same code twice. for me it's a good candidate for a private method

File: src/main/java/com/varukha/onlinebookstore/repository/book/BookSpecificationProviderManager.java [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+public class BookSpecificationProviderManager implements SpecificationProviderManager<Book> {
+    private final List<SpecificationProvider<Book>> bookSpecificationProviders;
+
+    @Override
+    public SpecificationProvider<Book> getSpecificationProvider(String key) {
+        return bookSpecificationProviders
+                .stream()
+                .filter(p -> p.getKey().equals(key))
+                .findFirst()
+                .orElseThrow(() -> new RuntimeException(
```
Comment (vovasalyha) [inline src/main/java/com/varukha/onlinebookstore/repository/book/BookSpecificationProviderManager.java:21]: better throw `NoSuchElementException`

**Review by vovasalyha — APPROVED** [review]
GJ. Please take a look on few suggestions.

**Pull request merged**